File: AdminPanel/AdminPanel.Web/Modules/ServerTypes/AdhomStoreCom/ProductForm.ts
import { StringEditor, DecimalEditor, LookupEditor, ImageUploadEditor, DateEditor, IntegerEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface ProductForm {
    Name: StringEditor;
    Description: StringEditor;
    Price: DecimalEditor;
    CatId: LookupEditor;
    Image: ImageUploadEditor;
    Type: StringEditor;
    SupllierName: StringEditor;
    EntryDate: DateEditor;
    ReviewUrl: StringEditor;
    Qty: IntegerEditor;
}

export class ProductForm extends PrefixedContext {
    static formKey = 'AdhomStoreCom.Product';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ProductForm.init)  {
            ProductForm.init = true;

            var w0 = StringEditor;
            var w1 = DecimalEditor;
            var w2 = LookupEditor;
            var w3 = ImageUploadEditor;
            var w4 = DateEditor;
            var w5 = IntegerEditor;

            initFormType(ProductForm, [
                'Name', w0,
                'Description', w0,
                'Price', w1,
                'CatId', w2,
                'Image', w3,
                'Type', w0,
                'SupllierName', w0,
                'EntryDate', w4,
                'ReviewUrl', w0,
                'Qty', w5
            ]);
        }
    }
}

File: AdminPanel/AdminPanel.Web/Modules/AdhomStoreCom/Product/ProductForm.cs
using Serenity.ComponentModel;
using Serenity.Web;
using System;

namespace AdminPanel.AdhomStoreCom.Forms
{
    [FormScript("AdhomStoreCom.Product")]
    [BasedOnRow(typeof(ProductRow), CheckNames = true)]
    public class ProductForm
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public int CatId { get; set; }
        public string Image { get; set; }
        public string Type { get; set; }
        public string SupllierName { get; set; }
        public DateTime EntryDate { get; set; }
        public string ReviewUrl { get; set; }
        public int Qty { get; set; }
    }
}